metrics_utility/automation_controller_billing/collectors.py

```
@@ -464,3 +480,56 @@ def main_host_table(since, full_path, until, **kwargs):
     return _copy_table(
         table='main_host', query=f'COPY ({query}) TO STDOUT WITH CSV HEADER', path=full_path, prepend_query=yaml_and_json_parsing_functions()
     )
+
+
+@register('total_workers_vcpu', '1.0', format='json', description=_('Total workers vCPU'))
+def total_workers_vcpu(since, full_path, until, **kwargs):
+    if 'total_workers_vcpu' not in get_optional_collectors():
+        return None
+
+    cluster_name = os.environ.get('METRICS_UTILITY_ANSIBLE_SAAS_CLUSTER_NAME')
+    if not cluster_name:
+        logger.error('environment variable METRICS_UTILITY_ANSIBLE_SAAS_CLUSTER_NAME is not set')
+        raise MissingRequiredEnvVar('environment variable METRICS_UTILITY_ANSIBLE_SAAS_CLUSTER_NAME is not set')
+
+    now = datetime.now(timezone.utc)
+
+    info = {'cluster_name': cluster_name, 'timestamp': now.isoformat(), 'nodes': []}
+
+    # If METRICS_UTILITY_VCPU_COUNT_ENABLED is not set or not set to true then it returns 1
+    vcpu_count_enabled_str = os.environ.get('METRICS_UTILITY_VCPU_COUNT_ENABLED')
+    vcpu_count_enabled = False
+    if vcpu_count_enabled_str and (vcpu_count_enabled_str.lower() == 'true'):
+        vcpu_count_enabled = True
+    if not vcpu_count_enabled:
+        return {'cluster_name': info['cluster_name'], 'total_workers_vcpu': '1'}
+
+    try:
+        kube_config.load_incluster_config()
+    except kube_config.ConfigException:
+        try:
+            kube_config.load_kube_config()
+        except kube_config.ConfigException as e:
+            logger.error(f'Could not configure Kubernetes Python client ERROR: {e}')
+            raise MetricsException(f'Could not configure Kubernetes Python client ERROR: {e}')
+
+    # Create a CoreV1Api client
+    api_instance = client.CoreV1Api()
+
+    nodes = api_instance.list_node()
```

It looks like here and below there are opportunities where we might fail silently trying to gather the nodes and then find the CPU value.